tree/treeplayer/test/leafs.cxx

```diff
@@ -234,3 +235,18 @@ TEST(TTreeReaderLeafs, MultipleReaders) {
    EXPECT_EQ(*v2, 1) << "Wrong value read for rv2!";
    EXPECT_EQ(*v3, 1) << "Wrong value read for rv3!";
 }
+
+// Test for https://github.com/root-project/root/issues/6881
+TEST(TTreeReaderLeafs, BranchAndLeafWithDifferentNames)
+{
+   TTree t("t", "t");
+   int x = 42;
+   t.Branch("x", &x, "y/I");
+   t.Fill();
+
+   TTreeReader r(&t);
+   TTreeReaderValue<int> rv(r, "x");
```

Comment:
```suggestion
   TTreeReaderValue<int> rv(r, "x");
   TTreeReaderValue<int> rvfull(r, "x.y");
```
and maybe 
```
   TTreeReaderValue<int> rvleaf(r, "y");
```

Reply:
`TreeReaderValue<int> rvleaf(r, "y");` does not work with branch name "x", leaf name "y":
```
Error in <TTreeReaderValueBase::CreateProxy()>: The tree does not have a branch called y. You could check with TTree::Print() for available branches.
```

is that ok?